matrix_client/client.py

```diff
@@ -603,6 +603,10 @@ def _sync(self, timeout_ms=30000):
             if room_id in self.rooms:
                 del self.rooms[room_id]
 
+        if self.encryption and 'device_one_time_keys_count' in response:
```

Comment:
Any chance we could formulate this as a listener that gets automatically added if `encryption=True` so we can keep all `_sync` actions other than simply updating local state in a single place?

Reply:
Mmmh, I'm not seeing how without adding a new listener loop, which seems pretty overkill for doing exactly the same thing differently.  
However, this is only the first sync extension of 3 (`device_list` and `to_device` are added later, and are non-blocking), so adding a loop might be worthwhile, although I'm not sure how clear it would look. Since we should get that right now, you may want to have a look at how `_sync` ends up looking at the end: https://github.com/matrix-org/matrix-python-sdk/pull/241/files#diff-6770c150117812114aea5a8aab1b7da9L554

Reply:
Ya. I see that listeners only work for events, and this is a top-level field in the response. Hm. After mulling it over, I think we should keep the listener system for user code.

I think this is probably fine for now, but could you add a `# TODO` comment above `_sync` mentioning this behavior so that when we get around to turning `_sync` into a proper public method with a docstring, we remember to document this?

Reply:
done